File: go/bowling_game/bowling_game/bowling_game_test.go
package bowling_game

import (
	"testing"
)

func TestAllzero(t *testing.T) {
	bg := BowlingGame{}

	for i := 0; i < 20; i++ {
		bg.Roll(0)
	}
	got := bg.Score()
	want := 0
	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func TestAllones(t *testing.T) {
	bg := BowlingGame{}

	for i := 0; i < 20; i++ {
		bg.Roll(1)
	}
	got := bg.Score()
	want := 20
	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func testOnepare(t *testing.T) {
	bg := BowlingGame{}

	bg.Roll(5)
	bg.Roll(5)
	bg.Roll(3)
	for i := 0; i < 17; i++ {
		bg.Roll(0)
	}
	got := bg.Score()
	want := 16
	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func testOnestrike(t *testing.T) {
	bg := BowlingGame{}

	bg.Roll(10)
	bg.Roll(3)
	bg.Roll(4)
	for i := 0; i < 16; i++ {
		bg.Roll(0)
	}
	got := bg.Score()
	want := 17
	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

func testPerfectgame(t *testing.T) {
	bg := BowlingGame{}

	for i := 0; i < 12; i++ {
		bg.Roll(10)
	}
	got := bg.Score()
	want := 300
	if got != want {
		t.Errorf("got %d, want %d", got, want)
	}
}

File: java_maven/BowlingGameKata/src/test/java/steelpipe/BowlingGameTest.java
package steelpipe;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class BowlingGameTest {

    @Test
    public void testAllZero() {
        BowlingGame bg = new BowlingGame();
        rollMany(bg, 20, 0);
        assertEquals(0, bg.score());
    }

    @Test
    public void testAllOnes() {
        BowlingGame bg = new BowlingGame();
        rollMany(bg, 20, 1);
        assertEquals(20, bg.score());
    }

    @Test
    public void testOneSpare() {
        BowlingGame bg = new BowlingGame();
        bg.roll(5);
        bg.roll(5);
        bg.roll(3);
        rollMany(bg, 17, 0);
        assertEquals(16, bg.score());
    }

    @Test
    public void testOneStrike() {
        BowlingGame bg = new BowlingGame();
        bg.roll(10);
        bg.roll(3);
        bg.roll(4);
        rollMany(bg, 16, 0);
        assertEquals(24, bg.score());
    }

    @Test
    public void testPerfectGame() {
        BowlingGame bg = new BowlingGame();
        rollMany(bg, 12, 10);
        assertEquals(300, bg.score());
    }

    private void rollMany(BowlingGame bg, int rolls, int pins) {
        for (int i = 0; i < rolls; i++) {
            bg.roll(pins);
        }
    }
}
